packages/odyssey-react-mui/src/ui-shell/NarrowUiShellContent.tsx

```diff
@@ -0,0 +1,404 @@
+  topNavBackgroundColor?: UiShellColors["topNavBackgroundColor"];
+}>(({ odysseyDesignTokens, isContentScrolled, topNavBackgroundColor }) => ({
+  alignItems: "stretch",
+  backgroundColor: topNavBackgroundColor,
+  boxShadow: isContentScrolled ? odysseyDesignTokens.DepthMedium : undefined,
+  clipPath: "inset(0 0 -100vh 0)",
+  display: "flex",
+  flexDirection: "column",
+  gridArea: "top-nav",
+  height: "100%",
+  justifyContent: "center",
+  position: "relative",
+  transition: `box-shadow ${odysseyDesignTokens.TransitionDurationMain} ${odysseyDesignTokens.TransitionTimingMain}`,
+  zIndex: UI_SHELL_BASE_Z_INDEX,
+}));
+
+const StyledTopNavMenu = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "topNavBackgroundColor",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  topNavBackgroundColor?: UiShellColors["topNavBackgroundColor"];
+}>(({ odysseyDesignTokens, topNavBackgroundColor }) => ({
+  alignItems: "center",
+  display: "flex",
+  gap: odysseyDesignTokens.Spacing3,
+  height: "100%",
+  justifyContent: "space-between",
+  maxHeight: TOP_NAV_HEIGHT,
+  minHeight: TOP_NAV_HEIGHT,
+  paddingBlock: odysseyDesignTokens.Spacing2,
+  paddingInline: odysseyDesignTokens.Spacing3,
+  transition: `box-shadow ${odysseyDesignTokens.TransitionDurationMain} ${odysseyDesignTokens.TransitionTimingMain}`,
+
+  ...(topNavBackgroundColor === odysseyDesignTokens.HueNeutralWhite &&
+    ({
+      borderBottomColor: odysseyDesignTokens.HueNeutral100,
+      borderBottomStyle: odysseyDesignTokens.BorderStyleMain,
+      borderBottomWidth: odysseyDesignTokens.BorderWidthMain,
+    } as CSSProperties)),
+}));
+
+const StyledTopNavSearch = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>(({ odysseyDesignTokens }) => ({
+  paddingBlock: odysseyDesignTokens.Spacing2,
+  paddingInline: odysseyDesignTokens.Spacing3,
+}));
+
+export type NarrowUiShellContentProps = Pick<HtmlProps, "testId"> &
+  Pick<UiShellNavComponentProps, "sideNavProps" | "topNavProps"> &
+  UiShellContentProps;
+
+const NarrowUiShellContent = ({
+  appContainerElement,
+  appContainerScrollingMode,
+  hasStandardAppContentPadding = true,
+  initialVisibleSections = ["TopNav", "SideNav", "AppSwitcher"],
+  onError = console.error,
+  optionalComponents,
+  sideNavProps,
+  topNavProps,
+}: NarrowUiShellContentProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const { isContentScrolled, scrollableContentRef: appContainerRef } =
+    useScrollState(appContainerElement);
+  const sideNavContainerRef = useRef<HTMLDivElement>(null);
+  const topNavContainerRef = useRef<HTMLDivElement>(null);
+
+  const uiShellContext = useUiShellContext();
+
+  const [isLeftSideMenuOpen, setIsLeftSideMenuOpen] = useState(false);
+  const [isRightSideMenuOpen, setIsRightSideMenuOpen] = useState(false);
+
+  const closeSideMenus = useCallback(() => {
```

Comment:
can we expose this or close the sidenav automatically on click of an item or add a prop/event so we can close the menus from the consuming app? In enduser when we select/click an item in the sidenav, the sidenav side collpase/hide so we can see the content.

Reply:
Good point. The only way to do that is by exposing a prop.